test/SdkCoinTest.t.sol

```diff
@@ -4,385 +4,602 @@ pragma solidity >=0.8.25 <0.9.0;
 // solhint-disable custom-errors,max-line-length
 import { Test } from "forge-std/Test.sol";
 import { SdkCoin } from "../src/utils/SdkCoin.sol";
-import { ERC20 } from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
 import { IISdkCoinErrors } from "../src/errors/IISdkCoinErrors.sol";
-
-// Discuss - Do we want to move mock contract to a mock folder?
-// Mock ERC20 token without the decimals function overridden
-contract MockERC20 is ERC20 {
-    constructor() ERC20("BaseToken", "BTK") {
-        _mint(msg.sender, 100_000_000_000_000_000_000); // Mint some tokens for testing
-    }
-}
-
-// Discuss - Do we want to move mock contract to a mock folder?
-// Mock ERC20 token with the decimals function overridden
-contract MockERC20Metadata is ERC20 {
-    uint8 private _decimals;
-
-    constructor(uint8 decimals_) ERC20("MetadataToken", "MTK") {
-        _decimals = decimals_;
-        _mint(msg.sender, 100_000_000_000_000_000_000); // Mint some tokens for testing
-    }
-
-    function decimals() public view override returns (uint8) {
-        return _decimals;
-    }
-}
-
-contract SdkCoinTest is Test {
+import { SafeCast } from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+import { MockERC20 } from "../test/mock/MockERC20.sol";
+import { MockERC20Metadata } from "../test/mock/MockERC20Metadata.sol";
+
+/**
+ * This test file validates the conversion functions between ERC20 token amounts and Cosmos SDK coin amounts,
+ * considering the differences in decimals. The testing strategy is twofold:
+ *
+ * 1. Table Testing:
+ *    - Involves creating a set of predefined test cases with specific inputs and expected outputs.
+ *    - Each test case is manually computed to cover various scenarios, including normal operation, edge cases,
+ *      and potential failure conditions.
+ *
+ * 2. Invariant Testing:
+ *    - Combines fuzz testing with invariant testing.
+ *    - Fuzz testing generates random inputs to uncover unexpected bugs and edge cases.
+ *    - Invariant testing ensures that core properties and conditions always hold true for the function's outputs
+ *      given random inputs.
+ *    - This approach provides robust validation without the need to manually compute expected values for each input,
+ *      enhancing the overall reliability and accuracy of the conversion functions.
+ */
+contract SdkCoinTest is Test, IISdkCoinErrors {
```

Comment:
ah, guys, do you prefer this file to be renamed `SdkCoinConversionsTest` and then we create separated files if we add more functions to the `SdkCoin` library?

Reply:
I think this is ok for now. We can split this into more files if and when we add more functionality